### PR TITLE
Add more documentation on development/contributing new submissions.

### PR DESCRIPTION
**Pull request by dlorenc**
# Changes

The goal of this commit is to add a set of "requirements" and guidelines
to serve as a starting point for maintainers reviewing new submissions.
I expect this will evolve over time as we get more experience maintaining
the catalog.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [X] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [X] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -1,3 +1,35 @@
...
+The Catalog repository is intended to serve as a location where users can find
+`Task`s and `Pipeline`s that are maintained, useful and follow established
+best practices.
+
+When reviewing PRs that add new `Task`s or `Pipeline`s, maintainers will follow
+the following guidelines:
+
+* Submissions should be independently usable.
+While this repository is meant to be educational, its primary goal is to serve
+as a place users can find, share and discover useful components.
```
Comment (kimsterv) [inline DEVELOPMENT.md]: I think an example here would be helpful.

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -1,3 +1,35 @@
...
+* Submissions should follow established best practices.
+Tekton is still young so this is going to be a shifting goalpost, but here are
+some examples:
+    * `Task`s should expose parameters and declare input/output resources, and
+    document them.
+    * Submissions should be as *portable* as possible.
+    They should not be hardcoded to specific repositories, clusters,
+    environments etc.
+    * Images should either be pinned by digest or point to tags that
```
Comment (sbwsg) [inline DEVELOPMENT.md]: s/tags that documented/tags with documented/

Comment (dlorenc) [reply DEVELOPMENT.md]: Thanks!

File: DEVELOPMENT.md (hunk 1 above)
Comment (dlorenc) [reply DEVELOPMENT.md]: How's this?

**Pull request merged**